sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEventListener.kt

```diff
@@ -0,0 +1,370 @@
+package io.sentry.android.okhttp
+
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.SpanStatus
+import okhttp3.Call
+import okhttp3.Connection
+import okhttp3.EventListener
+import okhttp3.Handshake
+import okhttp3.HttpUrl
+import okhttp3.Protocol
+import okhttp3.Request
+import okhttp3.Response
+import java.io.IOException
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.net.Proxy
+
+/**
+ *  Logs network performance event metrics to Sentry
+ *
+ *  Usage - add instance of [SentryOkHttpEventListener] in [OkHttpClient.eventListener]
+ *
+ * ```
+ * val client = OkHttpClient.Builder()
+ *     .eventListener(SentryOkHttpEventListener())
+ *     .addInterceptor(SentryOkHttpInterceptor())
+ *     .build()
+ * ```
+ *
+ * If you already use a [OkHttpClient.eventListener], you can pass it in the constructor.
+ *
+ * ```
+ * val client = OkHttpClient.Builder()
+ *     .eventListener(SentryOkHttpEventListener(myEventListener))
+ *     .addInterceptor(SentryOkHttpInterceptor())
+ *     .build()
+ * ```
+ */
+@Suppress("TooManyFunctions")
+class SentryOkHttpEventListener(
+    private val hub: IHub = HubAdapter.getInstance(),
+    private val originalEventListenerCreator: ((call: Call) -> EventListener)? = null
```

Comment:
Why is it called `original`?

Reply:
This is the listener that was (optionally) originally set by the user. We could name it in another way, like `delegateEventListener` if that's preferable